app/src/main/java/com/example/cosc341_project/data_classes/Post.java

```diff
@@ -0,0 +1,114 @@
+package com.example.cosc341_project.data_classes;
+
+import java.io.Serializable;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * <h4> Notes </h4>
+ * <p>
+ *     This is a data class for storing information related to posts. It is a supertype of <code>sightingPost</code>
+ * </p>
+ * <ul>
+ *     <li> All attributes are protected --- you cannot access them directly. </li>
+ *     <li>
+ *         All attributes have getters.
+ *         <b>Please do not modify the values or objects returned by the getters.</b>
+ *         Use the setter or other modification method built into the <code>Post</code> class.
+ *     </li>
+ *     <li>
+ *         Not all attributes have setters.
+ *         <ul>
+ *             <li> <code>UserId</code>, and <code>timestamp</code> are final (immutable) and initialized at construction. </li>
```

Comment:
Little unsure on how userId is determined. Assuming that would be a get statment in the post creation and then used on the post constructor.

Reply:
Yeah, sorry, to be clear `userId` is one of the ones passed as a parameter to the constructor. This is on my next to-do, but we will probably just choose one for our "current user" and you would just pass that every time.